fl4health/model_bases/autoencoders_base.py

```diff
@@ -0,0 +1,136 @@
+from abc import ABC, abstractmethod
+from typing import Optional, Tuple
+
+import torch
+import torch.nn as nn
+
+from fl4health.utils.dataset_converter import AutoEncoderDatasetConverter
+
+
+class AbstractAe(nn.Module, ABC):
+    """The base class for all Encoder-Decoder based models.
+    All we need to define such model is the type of the model, and the structure
```

Comment:
Super minor: "All we need to fine such model" -> "All we need to define for such models"